app/src/main/java/pm/gnosis/heimdall/ui/safe/main/SafeMainViewModel.kt

```diff
@@ -69,8 +76,25 @@ class SafeMainViewModel @Inject constructor(
             loadSelectedSafe()
         }
 
+    override fun observeSafe(safe: AbstractSafe): Flowable<Pair<String, String>> =
+        when(safe) {
+            is Safe -> safeRepository.observeSafe(safe.address)
+                .map { it.displayName(context) to it.address.shortChecksumString() }
+            is PendingSafe -> safeRepository.observePendingSafe(safe.hash)
+                .map { it.displayName(context) to it.address.shortChecksumString() }
+        }
```

Comment:
What do you think about moving the `address` to the `AbstractSafe` since it is common to both `Safe` and `PendingSafe`. 

Then we could also have:
```kotlin
when(safe) {
            is Safe -> safeRepository.observeSafe(safe.address)
            is PendingSafe -> safeRepository.observePendingSafe(safe.hash)
        }.map { it.displayName(context) to it.address.shortChecksumString() }
```

but really not important 😉 was more a question regarding the model itself

Reply:
I wanted to do that with address and name, but it is ugly with data classes.q